FlowCrypt/src/androidTest/java/com/flowcrypt/email/util/PrivateKeysManager.kt

```diff
@@ -64,6 +64,22 @@ class PrivateKeysManager {
       Thread.sleep(3000)
     }
 
+    fun savePubKeyToDatabase(assetsPath: String) {
+      savePubKeyToDatabase(getPgpKeyDetailsFromAssets(assetsPath))
+    }
+
+    fun savePubKeyToDatabase(pgpKeyDetails: PgpKeyDetails) {
+      val context = InstrumentationRegistry.getInstrumentation().targetContext
+      val email = requireNotNull(pgpKeyDetails.getPrimaryInternetAddress()).address.lowercase()
+      val roomDatabase = FlowCryptRoomDatabase.getDatabase(context)
+      if (roomDatabase.recipientDao().getRecipientByEmail(email) == null) {
+        roomDatabase.recipientDao().insert(requireNotNull(pgpKeyDetails.toRecipientEntity()))
+      }
+      roomDatabase.pubKeyDao().insert(pgpKeyDetails.toPublicKeyEntity(email))
+      // Added timeout for a better sync between threads.
+      Thread.sleep(500)
```

Comment:
maybe it should use lock instead?

Reply:
True. But if it's just a test, it may not be too bad - unless it does bring in actual flakiness.

Reply:
It's a temporary solution. `Thread.sleep()` over the whole code is a bad practice. But it can be fixed only after #1035